pkg/client/block/client.go

```diff
@@ -36,42 +39,147 @@ func NewBlockClient(
 		return nil, err
 	}
-	return &blockClient{eventsReplayClient: client}, nil
+
+	blockClient.forEachBlockEvent(ctx, latestBlockPublishCh)
+
+	if err := blockClient.getInitialBlock(ctx, latestBlockPublishCh); err != nil {
+		return nil, err
+	}
+
+	return blockClient, nil
 }
 
-// blockClient is a wrapper around an EventsReplayClient that implements the
+// blockReplayClient is a wrapper around an EventsReplayClient that implements the
 // BlockClient interface for use with cosmos-sdk networks.
-type blockClient struct {
+type blockReplayClient struct {
+	// onStartQueryClient is the RPC client that is used to query for the initial block
+	// upon blockClient construction. The result of this query is only used if it
+	// returns before the eventsReplayClient receives its first event.
+	onStartQueryClient cometclient.CometRPC
+
 	// eventsReplayClient is the underlying EventsReplayClient that is used to
 	// subscribe to new committed block events. It uses both the Block type
 	// and the BlockReplayObservable type as its generic types.
 	// These enable the EventsReplayClient to correctly map the raw event bytes
 	// to Block objects and to correctly return a BlockReplayObservable
 	eventsReplayClient client.EventsReplayClient[client.Block]
+
+	// latestBlockReplayObs is a replay observable that combines blocks observed by
+	// the block query client & the events replay client. It is the "canonical"
+	// source of block notifications for blockClient.
+	latestBlockReplayObs observable.ReplayObservable[client.Block]
+
+	// close is a function that cancels the context of the blockClient.
+	close context.CancelFunc
 }
 
 // CommittedBlocksSequence returns a replay observable of new block events.
-func (b *blockClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
+func (b *blockReplayClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
 	return b.eventsReplayClient.EventsSequence(ctx)
 }
 
-// LatestsNBlocks returns the last n blocks observed by the BockClient.
-func (b *blockClient) LastNBlocks(ctx context.Context, n int) []client.Block {
-	return b.eventsReplayClient.LastNEvents(ctx, n)
+// LastBlock returns the last blocks observed by the BlockClient.
+func (b *blockReplayClient) LastBlock(ctx context.Context) (block client.Block) {
+	// ReplayObservable#Last() is guaranteed to return at least one element.
+	return b.latestBlockReplayObs.Last(ctx, 1)[0]
 }
 
 // Close closes the underlying websocket connection for the EventsQueryClient
 // and closes all downstream connections.
-func (b *blockClient) Close() {
+func (b *blockReplayClient) Close() {
 	b.eventsReplayClient.Close()
+	//close(b.latestBlockPublishCh)
+	b.close()
+}
+
+// forEachBlockEvent asynchronously observes block event notifications from the
+// EventsReplayClient's EventsSequence observable & publishes each to latestBlockPublishCh.
+func (b *blockReplayClient) forEachBlockEvent(
```

Comment:
Wdyt of this instead?

```suggestion
func (b *blockReplayClient) asyncForwardBlockEvent(
```